pkg/gds/models/v1/models.go

```diff
@@ -175,6 +175,39 @@ func AppendCertReqID(vasp *pb.VASP, certreqID string) (err error) {
 	return nil
 }
 
+// DeleteCertReqID removes the certificate request ID from the VASP if it exists.
+func DeleteCertReqID(vasp *pb.VASP, certreqID string) (err error) {
+	// ID is required
+	if certreqID == "" {
+		return errors.New("cannot delete empty certificate request ID from extra")
+	}
+
+	// Unmarshal previous extra data.
+	extra := &GDSExtraData{}
+	if vasp.Extra != nil {
+		if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+			return fmt.Errorf("could not deserialize previous extra: %s", err)
+		}
+	} else {
+		extra.CertificateRequests = make([]string, 0)
+	}
+
+	// Search the slice for the certificate request ID
+	for i, containsID := range extra.CertificateRequests {
+		if certreqID == containsID {
+			// Remove the certificate request ID from the array
+			extra.CertificateRequests = append(extra.CertificateRequests[:i], extra.CertificateRequests[i+1:]...)
+			break
+		}
+	}
```

Comment:
If the list of CertificateRequests on VASPs gets longer, we may want to consider refactoring to an ordered list of UUIDs so that we can use `sort.Search` for better retrieval. Just something to keep in mind for the future, no action required now.